### DISCUSSION TITLE
Where to hold metadata for the collection?

### DISCUSSION THREAD

**Discussion by nfrmtk** [milvus-io/milvus#34000] [Q&A and General discussion] [answered]
Hello!
I have a following problem: when i create a collection, i need to set a password for it. Then for every insert i need to check if password given in insert request match the one I've set before. 
As I understood, this check cannot be performed with milvus, so i need an extra database to store passwords. Which one should it be?

**Comment by yhmo** [comment, marked as answer]
I think it is a client-side business. A client app fetches a password by collection's name, to decide whether this user can access this collection. Any database can do it. I think there are lots of options. MongoDb, Etcd, etc.

**Comment by xiaofan-luan** [comment]
you could store the password in kms, or system env(Might be risky, if some one can log in to your cluster)
